torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -799,8 +809,13 @@ std::shared_ptr<ProcessGroup::Work> ProcessGroupNCCL::collective(
   // Work itself will create the CUDA events on all GPUs of tensors
   auto work = initWork(devices);
 
-  // Store a reference to outputs to be used by WorkNCCL::getFuture.
-  work->outputs_ = std::make_shared<std::vector<at::Tensor>>(outputs);
+  // If single-process single-device mode, WorkNCCL::getFuture is supported.
+  // Store references to outputs and futureNCCLCallbackStream to be used by
+  // WorkNCCL::getFuture.
+  if (outputs.size() == 1) {
```

Comment:
I think it's fine to always do this instead of adding a condition here. We already have the appropriate check in FutureNCCL for this right?

Reply:
After defining FutureNCCL callback streams inside the constructor. I guess if I remove that check, the following will cause linter warning:
https://github.com/pytorch/pytorch/blob/d9c8b2e4613209f5f354df121fff61dadfc93afe/torch/lib/c10d/ProcessGroupNCCL.cpp#L828-L829

Reply:
What sort of linter error do you get? We should probably pass all the futureNCCLCallbackStreams_ to `WorkNCCL` as well and not just one. That way in the future it's easier to extend support for single process multi device mode.

Reply:
Agreed, I thought that if I do `outputs[0]` without any size check it would give a linter error. If we just pass a vector, it should be okay.